### ISSUE TITLE
ClassNotFoundException: play.libs.Json - can't use Diffson with Play's Java JSON library

### ISSUE THREAD

**Issue by kristian-lange** [gnieh/diffson#61]
I have a multi project Play application based on Play 2.5.11 and it's using Gnieh Diffson in one of the sub-projects. It's imported in the build.sbt with 
```
libraryDependencies ++= Seq(
  "org.gnieh" % "diffson-play-json_2.11" % "2.2.1"
)
```

It's a mixed project: Only the main-project and that Diffson-sub-project are using **Scala** - all other sub-projects using **Java**. 

Now, since I'm importing Diffson I'm getting this when I try to run the application:
```
Uncaught error from thread [application-akka.actor.default-dispatcher-9] shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[application]
java.lang.NoClassDefFoundError: play/libs/Json
	at play.mvc.StatusHeader.sendJson(StatusHeader.java:351)
...
Caused by: java.lang.ClassNotFoundException: play.libs.Json
```

It's not limited to the StatusHeader. Wherever I use Play's JSON library for Java (play.libs.Json - not play.api.libs.Json) I get a ClassNotFoundException for play.libs.Json. As soon as I remove the dependency to diffson-play-json it works again, although then I can't use Diffson anymore.

Do you have an idea how to use both, Diffson and play.libs.json, in parallel?

**Comment by satabin** [comment]
Hi,

It looks like the Java and Scala APIs use different packages according to the scaladoc and javadoc. It also looks both Java and Scala APIs are not in the same jar. The one diffson depends on is probably evicting the java one. On what Play! Json artifact are you depending without diffson?

**Comment by satabin** [comment]
Ok, I see you are using Play! 2.5.11 and diffson-play-json depends on Play! Json 2.6.0, so this dependency evicts yours because it has a higher version number. Moreover there are breaking changes between Play! 2.5 and 2.6 series so it may results in linking problems even if you get it to compile.

I would recommend either to update to Play! 2.6.x if you can or not to depend on diffson-play-json but rather on diffson-core (which is Json backend agnostic) and provide your own Json provider for diffson.
Supporting a new Json backend is quite straight-forward in diffson, and only requires a single class to be implemented. You can probably simply depends on `diffson-core` plus your Play! Json library  in version 2.5.11 and copy (maybe adapt a bit so that it compiles) the [diffson-play-json instance](https://github.com/gnieh/diffson/blob/master/playJson/src/main/scala/gnieh/diffson/PlayJsonInstance.scala) in your project. This is all what the `diffson-play-json` artifact does.

It is not ideal but I cannot afford to publish this single class artifact for all versions of Play!, cross-compiled with all supported versions of Scala and Java.

**Comment by kristian-lange** [comment]
Thank you very much Satabin, I saw those evict warning in my sbt log but thought they originate from somewhere else. I'll try to depend on diffson-core only. I'll try it today and give you feedback whether it's working.

**Comment by kristian-lange** [comment]
Actually I went back to Diffson version 2.1.0. This one works with Play 2.5. Later when I'm updating to Play 2.6 I'll switch back to Diffson 2.2. Thanks again!

**Comment by satabin** [comment]
Ok let me know if some fixes need to be backported to 2.1.x series.
